Test/patterns/PatternMatching.dfy

```diff
@@ -0,0 +1,89 @@
+// RUN: %dafny /compile:3 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*
+*  This file contains a collection of tests for features modified or introduced in PR 458
+*/
+
+
+datatype Alt = A(int) | B(int)
+datatype MyOption = Some(v: Alt) | None
+datatype MyPair = Pair(x:Alt, y:Alt)
```

Comment:
According to the Dafny style guide, use a space after each `:` (here and throughout this file)